api/src/main/java/com/google/common/flogger/LogSite.java

```diff
@@ -161,11 +162,19 @@ private InjectedLogSite(
         String internalClassName,
         String methodName,
         int encodedLineNumber,
-        @Nullable String sourceFileName) {
+        @Nullable String sourceFilePath) {
       this.internalClassName = checkNotNull(internalClassName, "class name");
       this.methodName = checkNotNull(methodName, "method name");
       this.encodedLineNumber = encodedLineNumber;
-      this.sourceFileName = sourceFileName;
+      this.sourceFileName = sourceFileName(sourceFilePath);
```

Comment:
Doing work in the constructor like this, rather than say, during bytecode injection, will cause massive amounts of additional runtime work in cases where a new injected log site is created per log method invocation. This is why I originally avoided doing any work in the constructor itself and let callers decide how best to avoid object allocations etc.

Reply:
Thanks, this wasn't the right place to do this, I'll fix it. I think a similar approach to `getClassName` would be appropriate?